engine/executor/dm/worker_test.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dm
+
+import (
+	"context"
+	"testing"
+
+	"github.com/go-mysql-org/go-mysql/mysql"
+	dmconfig "github.com/pingcap/tiflow/dm/dm/config"
+	"github.com/pingcap/tiflow/engine/jobmaster/dm/metadata"
+	"github.com/pingcap/tiflow/engine/lib"
+	"github.com/pingcap/tiflow/engine/lib/registry"
+	"github.com/pingcap/tiflow/engine/model"
+	dcontext "github.com/pingcap/tiflow/engine/pkg/context"
+	"github.com/pingcap/tiflow/engine/pkg/deps"
+	"github.com/pingcap/tiflow/engine/pkg/errors"
+	"github.com/pingcap/tiflow/engine/pkg/externalresource/broker"
+	extkv "github.com/pingcap/tiflow/engine/pkg/meta/extension"
+	kvmock "github.com/pingcap/tiflow/engine/pkg/meta/kvclient/mock"
+	pkgOrm "github.com/pingcap/tiflow/engine/pkg/orm"
+	"github.com/pingcap/tiflow/engine/pkg/p2p"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/dig"
+)
+
+type workerParamListForTest struct {
+	dig.Out
+
+	MessageHandlerManager p2p.MessageHandlerManager
+	MessageSender         p2p.MessageSender
+	FrameMetaClient       pkgOrm.Client
+	UserRawKVClient       extkv.KVClientEx
+	ResourceBroker        broker.Broker
+}
+
+// Init -> Poll -> Close
+func TestFactory(t *testing.T) {
+	cli, err := pkgOrm.NewMockClient()
+	require.NoError(t, err)
+	dctx := dcontext.Background()
+	dp := deps.NewDeps()
+	dctx = dctx.WithDeps(dp)
+	depsForTest := workerParamListForTest{
+		MessageHandlerManager: p2p.NewMockMessageHandlerManager(),
+		MessageSender:         p2p.NewMockMessageSender(),
+		FrameMetaClient:       cli,
+		UserRawKVClient:       kvmock.NewMetaMock(),
+		ResourceBroker:        broker.NewBrokerForTesting("exector-id"),
+	}
+	require.NoError(t, dp.Provide(func() workerParamListForTest {
+		return depsForTest
+	}))
+
+	// test factory
+	subtaskCfg := &dmconfig.SubTaskConfig{Name: "job-id", SourceID: "task-id", Flavor: mysql.MySQLFlavor}
+	content, err := subtaskCfg.Toml()
+	require.NoError(t, err)
+	RegisterWorker()
+	_, err = registry.GlobalWorkerRegistry().CreateWorker(dctx, lib.WorkerDMDump, "worker-id", "dm-jobmaster-id", []byte(content))
+	require.NoError(t, err)
+	_, err = registry.GlobalWorkerRegistry().CreateWorker(dctx, lib.WorkerDMLoad, "worker-id", "dm-jobmaster-id", []byte(content))
+	require.NoError(t, err)
+	_, err = registry.GlobalWorkerRegistry().CreateWorker(dctx, lib.WorkerDMSync, "worker-id", "dm-jobmaster-id", []byte(content))
+	require.NoError(t, err)
```

Comment:
seems we can test `messageAgent` is set.

Reply:
add at L88